compiler/src/middle_end/analyze_function_calls.re

```diff
@@ -0,0 +1,80 @@
+open Anftree;
+open Anf_iterator;
+open Grain_typed;
+
+let direct_function_calls = Ident_tbl.create(32);
+
+let register_function = id => {
+  Ident_tbl.add(direct_function_calls, id, 0);
+};
+
+let mark_function_call = id => {
+  switch (Ident_tbl.find_opt(direct_function_calls, id)) {
+  | Some(count) => Ident_tbl.replace(direct_function_calls, id, count - 1)
+  | None => ()
+  };
+};
+
+let mark_ident_seen = id => {
+  switch (Ident_tbl.find_opt(direct_function_calls, id)) {
+  | Some(count) => Ident_tbl.replace(direct_function_calls, id, count + 1)
```

Comment:
I think this +1/-1 logic could use a comment (since it's not immediately obvious what this table is tracking)